File: Stars Pattern/Reverse Hill/Pattern.py
#!/usr/bin/env python3

"""
Reverse Hill Pattern
"""

from sys import path

path.append("../.")


from ConsoleNumInput import inInt


#### Begin ####

n = inInt().getInput(5)

# Method 1
print("Method 1")

for a in range(n): 
    
    for b in range(a):
        print("   ", end="")
    
    for c in range(a, n):
        print(" * ", end="")
    
    for d in range(a + 1, n):
        print(" * ", end="")
    
    print("")


print("\n*~*~*~*~*~*~*~*~*~*~*~*\n")


# Method 2
print("Method 2")

for a in range(n):
    
    print("   " * a, end="")
    
    print(" * " * (n - a), end="")
    
    print(" * " * (n - a - 1))



print("")

"""
Author: Jovan De Guia
Github Username: jxmked
"""

File: Stars Pattern/Reverse Hill/Pattern.c
#include <stdio.h>
#include <stdlib.h>

// External File
#include "../ConsoleNumInput.c"

/**
 * Reverse Hill Pattern
 * */

static int loop(int n);


int main(void){
    // Default
    int n = 5;
    
    do {
        // From External File
        clrscr();
        
        loop(n);
        
        n = inInt();
        
        // This will never trigger. 
        // System required. Hahaha
        if(! n){
            return 0;
        }
    } while(1);
}

static int loop(int n){
    int i, j;
    
   // printf(" * \n");
    
    for(i = 0; i < n; i++){
        
        
        for(j = 0; j <= i; j++){
            printf("   ");
        }
        
        for(j = 1; j < (n - i) * 2; j++){
            printf(" * ");
        }
        
        printf("\n");
    }
    
    return 0;
}

/**
 * Written by Jovan De Guia
 * Github Username: jxmked
 * */
